satpy/readers/clavrx.py

```diff
@@ -94,9 +108,29 @@ def _get_rows_per_scan(sensor: str) -> Optional[int]:
     return None
 
 
+def _scale_data(data_arr: Union[xr.DataArray, int], scale_factor: float, add_offset: float) -> xr.DataArray:
+    """Scale data, if needed."""
+    scaling_needed = not (scale_factor == 1.0 and add_offset == 0.0)
+    if scaling_needed:
+        data_arr = data_arr * scale_factor + add_offset
```

Comment:
Do you/we need `np.float32(scale_factor)` (and on the offset) to make sure 64-bit floats aren't created unnecessarily?

Reply:
Yes.